src/diagnostics/core_diagnostics.jl

```diff
@@ -733,3 +733,315 @@ add_diagnostic_variable!(
     comments = "Elevation of the horizontal coordinates",
     compute! = compute_orog!,
 )
+
+###
+# cloud liquid water specific humidity (1d)
+###
+compute_clwvi!(out, state, cache, time) =
+    compute_clwvi!(out, state, cache, time, cache.atmos.moisture_model)
+compute_clwvi!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("clwvi", model)
+
+function compute_clwvi!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* cache.precomputed.cloud_diagnostics_tuple.q_liq,
+        )
+        return out
+    else
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* cache.precomputed.cloud_diagnostics_tuple.q_liq,
+        )
+    end
+end
```

Comment:
Let's add `lwp` and `clivi` as well. (I misread the comment for `clwvi` before, it was liquid + ice water path. `lwp` is liquid water path, and `clivi` is ice water path.)